docs/pages/docs/apis/access-control.mdx

```diff
@@ -20,244 +20,171 @@ export default config({
+import { checkbox } from '@keystone-next/keystone/fields';
 
 export default config({
   lists: createSchema({
     ListKey: list({
-      // Imperative access control definition
-      access: args => true,
+      fields: {
+        isReadable: checkbox(),
+        isUpdatable: checkbox(),
+        isDeletable: checkbox(),
+      }
+      access: {
+        filter: {
+          query: ({ session, context, listKey, operation }) => {
+            return { isReadable: { equals: true } };
+          },
+          update: ({ session, context, listKey, operation }) => {
+            return { isUpdatable: { equals: true } };
+          },
+          delete: ({ session, context, listKey, operation }) => {
+            return { isDeletable: { equals: true } };
+          },
+        }
+      },
     }),
   }),
 });
 ```
 
-#### Imperative Function Arguments
-
-Imperative access control functions are passed a collection of arguments which can be used to determine whether the operation is allowed.
-
-| Argument        | Description                                                                                                                                               |
-| --------------- | --------------------------------------------------------------------------------------------------------------------------------------------------------- |
-| `listKey`       | The key of the list being operated on.                                                                                                                    |
-| `operation`     | The CRUD operation being performed (`'create'`, `'read'`, `'update'`, `'delete'`).                                                                        |
-| `session`       | The current session object. See the [Sessions API](./session) for details.                                                                                |
-| `originalInput` | For `create` and `update` operations, this is the value of `data` passed into the mutation. For `read` and `delete` operations this value is `undefined`. |
-| `itemId`        | The `id` of the item being updated/deleted in `update` and `delete` operations. `undefined` for other operations.                                         |
-| `context`       | The [`KeystoneContext`](./context) object of the originating GraphQL operation.                                                                           |
-
-```typescript
-import { config, createSchema, list } from '@keystone-next/keystone';
+!> Filter access control cannot be used on `create` operations, as there is no pre-existing item to filter against.
 
-export default config({
-  lists: createSchema({
-    ListKey: list({
-      // Imperative access control definition
-      access: ({
-        listKey,
-        operation,
-        session,
-        originalInput,
-        itemId,
-        context,
-      }) => {
-        return true;
-      },
-    }),
-  }),
-});
-```
+### Item (mutations only)
```

Comment:
Is why this is mutation only documented somewhere?

Reply:
```
?> Item-level access control is not available for `query` operations.
Applying access control after fetching items would lead to inconsistent pagination behaviour and incorrect `count` results.
```